syntax = "proto3";
package models;
import "proto/paymentrequest.proto";

// Message sent from service to BIP 70 server
message InvoiceParams {
    // Bitcoin network
    string network = 1;
    // Amount
    uint64 amount = 2;
    // Issue time
    uint64 time = 3;
    // Expiry time
    uint64 expires = 4;
    string memo = 5;
    bytes merchant_data = 6;
    // Memo to be sent in PaymentAck
    string memo_ack = 7;
    // Token to be signed and put in the payment ack headers
    bytes token = 8;
    // Data to be inserted into op_return
    bytes tx_data = 9;
    // Callback URL
    string callback_url = 10;
}

// Message sent in response to the InvoiceParams
message InvoiceResponse {
    // ID of the payment
    uint64 payment_id = 1;
    // Payment request to be sent to customer
    PaymentRequest payment_request = 2;
}

// Message sent to service on successful payment
message CallbackPayload {
    uint64 payment_id = 1;
    PaymentACK payment_ack = 2;
}
